src/awkward/_typetracer.py

```diff
@@ -704,6 +704,42 @@ class TypeTracer(NumpyLike):
     known_data: Final = False
     known_shape: Final = False
 
+    class _FakeModule:
+        def __getattr__(self, function_name):
+            ufunc = getattr(numpy, function_name, None)
+            if isinstance(ufunc, numpy.ufunc):
+                return self._generic_ufunc_func(ufunc)
+            else:
+                raise ak._errors.wrap_error(
+                    AttributeError(
+                        "only ufuncs have automatic TypeTracer implementations"
+                    )
+                )
+
+        @staticmethod
+        def _generic_ufunc_func(ufunc):
+            def generic_ufunc(*inputs, **kwargs):
+                for x in inputs:
+                    getattr(x, "touch_data", lambda: None)()
+
+                replacements = []
+                to_broadcast = []
+                for x in inputs:
+                    if hasattr(x, "dtype") and hasattr(x, "shape"):
+                        replacements.append(numpy.empty(0, x.dtype))
+                        to_broadcast.append(x)
+                    else:
+                        replacements.append(x)
+
+                broadcasted = TypeTracer.instance().broadcast_arrays(*to_broadcast)
+
+                result = ufunc(*replacements, **kwargs)
+                return TypeTracerArray(result.dtype, shape=broadcasted[0].shape)
```

Comment:
This is raising an `AttributeError` in e.g. `ak.std`:
```python
>>> import awkward as ak
>>> a = ak.Array([[1, 2, 3], [4, 5]])
>>> tt = a.layout.to_typetracer(forget_length=True)
>>> tt = ak.Array(tt)
>>> tt
<Array-typetracer [...] type='?? * var * int64'>
>>> ak.std(tt, axis=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ddavis/software/repos/awkward/src/awkward/operations/ak_std.py", line 84, in std
    return _impl(x, weight, ddof, axis, keepdims, mask_identity)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ddavis/software/repos/awkward/src/awkward/operations/ak_std.py", line 172, in _impl
    return ak._nplikes.nplike_of(x, weight).sqrt(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ddavis/software/repos/awkward/src/awkward/_nplikes.py", line 332, in sqrt
    return self._module.sqrt(x, out=maybe_out)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ddavis/software/repos/awkward/src/awkward/_typetracer.py", line 737, in generic_ufunc
    return TypeTracerArray(result.dtype, shape=broadcasted[0].shape)
                           ^^^^^^^^^^^^
  File "/Users/ddavis/software/repos/awkward/src/awkward/highlevel.py", line 1125, in __getattr__
    raise ak._errors.wrap_error(AttributeError(f"no field named {where!r}"))
AttributeError: while calling

    ak.std(
        x = <Array-typetracer [...] type='?? * var * int64'>
        weight = None
        ddof = 0
        axis = 1
        keepdims = False
        mask_identity = False
    )

Error details: no field named 'dtype'. Did you mean: 'type'?
```

Reply:
That's quick! I'll take a look

Reply:
OK, I'm fairly confident this is because we're using Awkward's ufunc dispatching, as you're operating on the high level object. The two are distinct, and I'm touching this code in another PR. Let me spit that up so we can fix this bug.